lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -253,16 +248,21 @@ def build_request_url(action, id = nil)
         base_url + endpoint
       end
 
-      def success_from(response)
-        SUCCESS_MESSAGES.include?(response.dig('operation', 'operationResult'))
+      def success_from(action, response)
+        case action
+        when :capture, :refund
+          response.include?('operationId') && response.include?('operationTime')
+        else
+          SUCCESS_MESSAGES.include?(response.dig('operation', 'operationResult'))
+        end
       end
 
       def message_from(response)
         response.dig('operation', 'operationResult') || response.dig('errors', 0, 'description')
       end
 
       def authorization_from(response)
```

Comment:
💬 Comment

I get the intention of the guard clause, what about setting the default value for `response` as {}, in that way you don't need the guard clause.

Reply:
Done!